javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,39 @@
+var restart = document.querySelector('button');
+var typer = document.getElementById('phrase');
+var arr = typer.textContent.split('');
+var spanArr = [];
+var position =  0;
+
+for(var i = 0; i < arr.length; i++){
+  var spanCharacter = document.createElement('span');
+  spanCharacter.textContent = arr[i];
+  spanArr.push(spanCharacter);
+}
+
+window.addEventListener('keyup', function(e) {
+  if (arr[position].toLowerCase() === e.key) {
+
+    spanArr[position].classList.remove('incorrect');
+    spanArr[position].classList.add('correct');
+
+    spanArr[position].classList.remove('underline');
+    spanArr[position+1].classList.add('underline');
+
+    typer.innerHTML = '';
+
+    for(var i = 0; i < arr.length; i++){
+      typer.appendChild(spanArr[i]);
+    }
```

Comment:
You shouldn't need to destroy all elements and rebuild them every time a letter is typed. Instead, when you generate your array of spans when the page loads, you should append those to the DOM right away and just update the class list on the current letter whenever the user types.